### PR TITLE
Graveyard update take two

### PR DESCRIPTION
**Pull request by Rhials**
## About The Pull Request

**PR body copied from last PR (#83149). I fucked something up (I think by leaving dream maker open while trying to fix the merge conflicts?) and rather than try and walk backwards I'm just making a new branch.**

This implements the digging of graves on most soil/dirt/planetary type turfs, and gives the coroner their own private burial ground.

**Change 1 - Gravedigging:**

You can right-click planetary/dirt/grass tiles using a shovel or shovel subtype (or entrenching tool). Speed varies on the type of shovel you are using. This creates a Makeshift Grave, an unmarked burial mound (different from the ones at the elephant graveyard).

This is handled through the new gravedigger component, which is mostly unremarkable but worth mentioning in case anyone wants to add this behavior elsewhere.

**Change 2 - Icebox Morgue Graveyard:**

image

The icebox morgue now has private burial ground, sealed off by a fence. Mourners are expected to keep out and perform their grieving at the appropriate distance. This does not affect the chaplain's burial ground, which is publicly accessible from the outside. This gives a more secure place to bury bodies (I'm sure someone will have a reason for this some day) and may lead to fighting over corpses, which I think is funny.

Also, there might be some goodies left in those graves, but you wouldn't go graverobbing just for some useless loot, would you??

This also adds a new area type, graveyard, which is mostly just the icemoon outdoors with the spooky ambiance of the morgue.

image
## Why It's Good For The Game

Now you can bury your friends in an unmarked grave! Bury people alive! Bury your treasure, or reminders of the sins you've committed! Bury anything, anywhere you want!

The morgue graveyard is a nice bit of flavor. I know the Chaplain already gets one (I forgot this when I started this PR though) but the Coroner is an equal-if-not-more-important corpsekeeper than them. 
## Changelog
:cl: Rhials
add: Shovels and entrenching tools can be used to dig graves on asteroid/dirt/etc. surfaces. Neat!
add: The Icebox Morgue has been given a fenced-off graveyard in the back.
code: burn_tile() is no longer double-defined on asteroid turfs.
/:cl:

### CODE REVIEW

**Review by L0pz — APPROVED** [review]
Yes please I'd like to bury a chest full of treasures I've tided

**Review by Maurukas — COMMENTED** [review]
Please push an empty commit or merge master and push to force Mapdiffbot to rerun. It has been fixed.

**Review by Maurukas — APPROVED** [review]
Mapping's fine. And I like it.

Maybe Lavaland could use one as well.

File: code/modules/mining/equipment/mining_tools.dm [hunk 1]
```diff
@@ -135,6 +135,8 @@
 	attack_verb_continuous = list("bashes", "bludgeons", "thrashes", "whacks")
 	attack_verb_simple = list("bash", "bludgeon", "thrash", "whack")
 	sharpness = SHARP_EDGED
+	///How quickly do we dig graves?
+	var/grave_speed = 8 SECONDS
```
Comment (Jacquerel) [inline code/modules/mining/equipment/mining_tools.dm:139]: This var doesn't do anything

**Review by Jacquerel — CHANGES_REQUESTED** [review]

**Pull request merged**